# HG changeset patch
# User John Donoghue <john.donoghue@ieee.org>
# Date 1426210337 14400
#      Thu Mar 12 21:32:17 2015 -0400
# Node ID 29fe196d2ce05ef953bc9c181a5b17ef4315df20
# Parent  2d5cbb6ac845dcb71b71602d4627131446a2ed8c
GUI: use reource manager for command window icons (Bug #44528)

* libgui/qterminal/libqterminal/QTerminal.h
  (toplevel): include resource-manager.h
  (QTerminal::QTerminal): Use resource manager icons for copy and paste menus

* libgui/qterminal-module.mk:
  (qterminal_libqterminal_la_CPPFLAGS): add -I libgui/src

diff --git a/libgui/qterminal-module.mk b/libgui/qterminal-module.mk
--- a/libgui/qterminal-module.mk
+++ b/libgui/qterminal-module.mk
@@ -30,17 +30,18 @@ noinst_HEADERS += \
 qterminal_libqterminal_la_MOC = \
   qterminal/libqterminal/moc-QTerminal.cc
 
 nodist_qterminal_libqterminal_la_SOURCES = $(qterminal_libqterminal_la_MOC)
 
 qterminal_libqterminal_la_CPPFLAGS = \
   $(AM_CPPFLAGS) \
   @QT_CPPFLAGS@ \
-  -I$(srcdir)/qterminal/libqterminal
+  -I$(srcdir)/qterminal/libqterminal \
+  -I$(srcdir)/src
 
 qterminal_libqterminal_la_CFLAGS = $(AM_CFLAGS)
 
 qterminal_libqterminal_la_CXXFLAGS = $(AM_CXXFLAGS)
 
 if WIN32_TERMINAL
 
 qterminal_libqterminal_la_SOURCES = \
diff --git a/libgui/qterminal/libqterminal/QTerminal.h b/libgui/qterminal/libqterminal/QTerminal.h
--- a/libgui/qterminal/libqterminal/QTerminal.h
+++ b/libgui/qterminal/libqterminal/QTerminal.h
@@ -30,16 +30,18 @@ see <http://www.gnu.org/licenses/>.
 #include <QStringList>
 #include <QColor>
 #include <QList>
 #include <QMenu>
 #include <QClipboard>
 #include <QApplication>
 #include <QAction>
 
+#include "resource-manager.h"
+
 class QTerminal : public QWidget
 {
   Q_OBJECT
 
 public:
 
   static QTerminal *create (QWidget *xparent = 0);
 
@@ -122,21 +124,21 @@ protected:
   QTerminal (QWidget *xparent = 0) : QWidget (xparent)
   {
     // context menu
     setContextMenuPolicy (Qt::CustomContextMenu);
 
     _contextMenu = new QMenu (this);
 
     _copy_action = _contextMenu->addAction (
-                     QIcon::fromTheme ("edit-copy", QIcon (":/actions/icons/edit-copy.svg")),
+                     resource_manager::icon ("edit-copy"),
                      tr ("Copy"), this, SLOT (copyClipboard ()));
 
     _paste_action = _contextMenu->addAction (
-                     QIcon::fromTheme ("edit-paste", QIcon (":/actions/icons/edit-paste.svg")),
+                     resource_manager::icon ("edit-paste"),
                       tr ("Paste"), this, SLOT (pasteClipboard ()));
 
     _contextMenu->addSeparator ();
 
     _selectall_action = _contextMenu->addAction (
                       tr ("Select All"), this, SLOT (selectAll ()));
 
 
